app/controllers/api/v3/private/user_controller.rb

```diff
@@ -0,0 +1,13 @@
+module API
+  module V3
+    module Private
+      class Api::V3::Private::UserController < ::ApplicationController
+        include JwtAuthorizationConcern
+
+        def index
+          render json: { username: @current_user.id }
```

Comment:
It may be worthwhile to separate the username from the user's ID in the database. That would make it easier to change the username, and add a way to refer to a user without using the username.

Reply:
if we do move to a single auth service for our stuff, username makes the most sense as the key.  I don't think allowing changes to usernames is a great idea, though, so i was not currently worrying about it.

are you more worried about that or "ints are smaller than strings" which i don't think is gonna be a serious concern for us.

Reply:
I'm thinking about conceptual coupling rather than foreign key performance. I'm suggesting we could decouple the internal user ID from any identifiers in the authentication system.

When a user authenticates, we could look up the username with an index, and the internal ID / primary key would be a separate field.

If we use the username as a foreign key it makes it much harder to change our mind if we want to allow username changes.

There's also a case where we accidentally allow a username change in the auth system, and the user loses access to their decks.

Someone else would also be able to register a user with that username and get their decks.

I don't think this affects how easy it is to migrate between auth systems. There is a separate blocker for username changes in that the username is the only identifier we take from the auth system, so we don't have a reference that will be the same after the change. That's a much easier problem to solve if we wanted to, as long as the username is only in one place in the DB.

If we index the auth system identifiers alongside the internal ID, we could even include both the username and the external ID from the auth system, and which system that ID is from. That way we could avoid loss of access both when we migrate between auth systems and when the username changes. If we look up the user by the external ID instead of the username, we could also avoid unauthorised access if the username does change in the auth system.

These are all weird corner cases but I think a clean solution can limit the cognitive load of thinking about those things, as well as keeping the interactions simple.

Reply:
we don't allow username changes, so I don't think that risk is serious.  Moreover, i don't think we *want* to allow username changes.  The new auth system is going to be seeded with current NRDB info, not just wide open.

dealing with potential email hijacks or other things could still prove destructive, but not because of a schema choice here.

I think we can deal with this a bit later if we need to revisit it.  Are you OK getting this in to get some foundations laid down?

I can

Reply:
Sure, that's fine.